src/components/WalletInit/ConnectNanoX/ConnectNanoXScreen.js

```diff
@@ -1,284 +1,100 @@
 import {injectIntl, defineMessages, intlShape} from 'react-intl'
 import {compose} from 'redux'
-import TransportBLE from '@ledgerhq/react-native-hw-transport-ble'
-import {ErrorCodes} from '@cardano-foundation/ledgerjs-hw-app-cardano'
-import {BleError} from 'react-native-ble-plx'
+import {withHandlers} from 'recompose'
 
-import {
-  getHWDeviceInfo,
-  BluetoothDisabledError,
-  GeneralConnectionError,
-  LedgerUserError,
-} from '../../../crypto/byron/ledgerUtils'
-import {Text, BulletPointItem, ProgressStep} from '../../UiKit'
+import LedgerConnect from '../../Ledger/LedgerConnect'
+import {getHWDeviceInfo} from '../../../crypto/byron/ledgerUtils'
+import {ProgressStep} from '../../UiKit'
 import {withNavigationTitle} from '../../../utils/renderUtils'
-import DeviceItem from './DeviceItem'
 import {WALLET_INIT_ROUTES} from '../../../RoutesList'
-import {ledgerMessages} from '../../../i18n/global-messages'
+import {Logger} from '../../../utils/logging'
 
 import styles from './styles/ConnectNanoXScreen.style'
-import image from '../../../assets/img/bluetooth.png'
 
 import type {ComponentType} from 'react'
 import type {Device} from '@ledgerhq/react-native-hw-transport-ble'
 import type {Navigation} from '../../../types/navigation'
+import type {DeviceId, DeviceObj} from '../../../crypto/byron/ledgerUtils'
 
 const messages = defineMessages({
   title: {
     id: 'components.walletinit.connectnanox.connectnanoxscreen.title',
-    defaultMessage: '!!!Connect to Ledger',
-  },
-  caption: {
-    id: 'components.walletinit.connectnanox.connectnanoxscreen.caption',
-    defaultMessage: '!!!Scanning bluetooth devices...',
-  },
-  introline: {
-    id: 'components.walletinit.connectnanox.connectnanoxscreen.introline',
-    defaultMessage: "!!!You'll need to:",
+    defaultMessage: '!!!Connect to Ledger Device',
   },
   exportKey: {
     id: 'components.walletinit.connectnanox.connectnanoxscreen.exportKey',
     defaultMessage:
       '!!!Action needed: Please, export public key from your Ledger device.',
   },
-  error: {
-    id: 'components.walletinit.connectnanox.connectnanoxscreen.error',
-    defaultMessage:
-      '!!!An error occurred while trying to connect with your hardware wallet:',
-  },
 })
 
-const deviceAddition = (device) => ({devices}) => ({
-  devices: devices.some((i) => i.id === device.id)
-    ? devices
-    : devices.concat(device),
-})
+const _navigateToSave = async (
+  deviceId: ?DeviceId,
+  deviceObj: ?DeviceObj,
+  navigation: Navigation,
+): Promise<void> => {
+  Logger.debug('deviceId', deviceId)
+  Logger.debug('deviceObj', deviceObj)
+  const useUSB = navigation.getParam('useUSB') === true
+  if (deviceId == null && deviceObj == null) {
+    throw new Error('null descriptor, should never happen')
+  }
+  const hwDeviceInfo = await getHWDeviceInfo(deviceId, deviceObj, useUSB)
+  navigation.navigate(WALLET_INIT_ROUTES.SAVE_NANO_X, {hwDeviceInfo})
+}
 
-type Props = {|
+type Props = {
```

Comment:
Not sure why you made this prop type no longer exact

Reply:
I can't have them exact because the `withHandlers` HOC does an inexact prop merge :/